frontend/src/app/filter/base-filter.component.ts

```diff
@@ -25,6 +26,12 @@ export abstract class BaseFilterComponent<FilterData> implements OnChanges {
 
     ngOnChanges(changes: SimpleChanges): void {
         if (changes.filter) {
+            if (this.filterSubscription) {
+                this.filterSubscription.unsubscribe();
+            }
+            this.filterSubscription = this.filter.update.subscribe(() =>
+                this.onFilterSet(this.filter)
+            );
```

Comment:
This has the appearance of creating an update loop. It doesn't, because the SearchFilter class will ignore updates with the same data as its current state. It would be nice to add a comment that explains that.